python/meep.i

```diff
@@ -81,6 +84,78 @@ static double py_callback_wrap(const meep::vec &v) {
   $1 = PyCallable_Check($input);
 }
 
+// Typemap suite for vector3
+
+%typemap(in) vector3 {
+    if(!pyv3_to_v3($input, &$1)) {
+        SWIG_fail;
+    }
+}
+
+// Typemap suite for GEOMETRIC_OBJECT
+
+%typecheck(SWIG_TYPECHECK_POINTER) GEOMETRIC_OBJECT {
+    PyObject *geom_mod = PyImport_ImportModule("geom");
+    PyObject *geom_dict = PyModule_GetDict(geom_mod);
+    PyObject *py_sphere = PyDict_GetItemString(geom_dict, "GeometricObject");
+    $1 = PyObject_IsInstance($input, py_sphere);
+    Py_XDECREF(geom_mod);
```

Comment:
Don't you need to decref the other two objects?  And isn't `py_sphere` a misnomer here, maybe you should rename to `py_geometricobject`?

I don't think you need `geom_dict`.  You can do
```c
py_geometricobject = PyObject_GetAttrString(geom_mod, "GeometricObject");
```
directly.

To speed this up you could potentially cache `py_geometricobject` in a global.  That is, define a function:
```
static PyObject *py_geometricobject(void) {
    static PyObject *geometricobject = NULL;
    if (geometric_object == NULL) {
        PyObject *geom = PyImport_ImportModule("geom");
        geometricobject = PyObject_GetAttrString(geom_mod, "GeometricObject");
        Py_XDECREF(geom);
    }
    return geometricobject;
}
```
and then just do `PyObject_IsInstance($input, py_geometricobject())`.